### PR TITLE
WhileLoopFactorial: Optimize and add tests

### PR DESCRIPTION
**Pull request by merelymyself**
# Welcome to the JavaScript community

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [ ] Add an algorithm?
- [x] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Maths/WhileLoopFactorial.js [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-  // Step 1. variable result to store num
-  let result = num
-  // If num = 0 OR 1, the factorial will return 1
-  if (num === 0 || num === 1) { return 1 }
+  // Step 1. Handles cases where num is 0 or 1, by returning 1.
+  let result = 1
   // Step 2. WHILE loop
   while (num > 1) {
-    num-- // decrement 1 at each iteration
     result = result * num // or result = result * num;
```
Comment (appgurueu) [inline Maths/WhileLoopFactorial.js:9]: Could make this a `result *= num`

**Review by appgurueu — APPROVED** [review]
I'd probably prefer a `for`-loop but this is fine. Thanks for adding tests in particular. @raklaptudirm do we have a guideline on authorship documented as comments in code files? I believe it should be kept only in the Git history and the comment should be removed entirely.

File: Maths/test/WhileLoopFactorial.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { factorialize } from '../WhileLoopFactorial'
+
+test('Testing on 3!', () => {
+  const three_factorial = factorialize(3)
```
Comment (appgurueu) [inline Maths/test/WhileLoopFactorial.test.js:4]: Your identifiers should be in camel case. You also don't need the variables (you should just do `expect(factorialize(3)).toBe(6)`).

**Comment by appgurueu** [comment]
> standard: Use JavaScript Standard Style (https://standardjs.com/)
standard: Run `standard --fix` to automatically fix some problems.
  /home/runner/work/Javascript/Javascript/Maths/test/WhileLoopFactorial.test.js:3:23: Missing space before function parentheses.
  
@raklaptudirm space before function parentheses is bad style IMO, can't standard be configured differently?

**Comment by raklaptudirm** [comment]
@appgurueu No standard is a style that can't be configured. That is basically the reason we use it, as it is easy of beginners. Imo, as long a consistent style is being maintained the details don't matter.

File: Maths/test/WhileLoopFactorial.test.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
 import { factorialize } from '../WhileLoopFactorial'
 
 function testFactorial(n, expected) {
```
Comment (appgurueu) [inline Maths/test/WhileLoopFactorial.test.js:3]: You were supposed to add the space here, after `testFactorial`, not after all function calls...

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**